packages/frontend/model/extract-nav.ts

```diff
@@ -102,6 +102,40 @@ export const extract = (data: {}): NavType => {
                     '',
                 ),
             },
+            ampHeader: {
+                contribute: getString(
+                    data,
+                    'config.readerRevenueLinks.ampHeader.contribute',
+                    '',
+                ),
+                subscribe: getString(
+                    data,
+                    'config.readerRevenueLinks.ampHeader.subscribe',
+                    '',
+                ),
+                support: getString(
+                    data,
+                    'config.readerRevenueLinks.ampHeader.support',
+                    '',
+                ),
+            },
+            ampFooter: {
+                contribute: getString(
+                    data,
+                    'config.readerRevenueLinks.ampFooter.contribute',
+                    '',
+                ),
+                subscribe: getString(
+                    data,
+                    'config.readerRevenueLinks.ampFooter.subscribe',
+                    '',
+                ),
+                support: getString(
+                    data,
+                    'config.readerRevenueLinks.ampFooter.support',
+                    '',
+                ),
+            },
```

Comment:
Ought these be `nonEmptyString`s, do you think? Should we fail to render if no URL is provided from frontend?

Reply:
I guess I set a precedence by using `getString` with an empty string fallback for the header/footer/sideMenu reader revenue links. Returning a 500 if for whatever reason frontend fails to provide anything seemed quite strict at the time but if reader revenue links are up there with required content in terms of importance then we shouldn't be permissive about their presence.

If we're happy to be permissive then we should put a check in the component and not render the link if the values of these properties are empty strings, which is something I don't think we're actually doing in the web header! Does that sound fair?

Reply:
Yeah, I prefer the permissive approach. 

I raised it here because I don't think we've discussed how strictly we enforce the expected data model.

I'll add a Trello card to discuss it more thoroughly after Data Model v2 is merged (guardian/frontend#20905)